### PR TITLE
Cal 34961 no shows

### PR DESCRIPTION
**Pull request by RSalaam**
# Changes
- WIP to create and render no-show data for a scheduled event
(I'm not sure if markAsNoShow needs to accept invitee uri to pass to postRequestConfiguration. The server-side API seems to overlap too much with the fetch request in the component.)

# Story
https://calendlyapp.atlassian.net/browse/CAL-34961

### CODE REVIEW

File: components/invitee.jsx [hunk 1]
```diff
@@ -2,60 +2,114 @@ import React, { useState, useEffect } from 'react';
...
+            <th>E-mail</th>
+            <th>Date/time at scheduling</th>
+            <th>Questions + Answers</th>
+            <th>Rescheduled</th>
+            <th>Timezone</th>
+          </tr>
+        </thead>
+        <tbody>
+          {invitees.map((invitee, i) => (
+            <tr key={i}>
```
Comment (tristanoneil) [inline components/invitee.jsx:76]: It's not ideal to use an index as `key` w/ React. See [here](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318). I would use something else unique like the `uri`.

File: routes/api.js [hunk 2]
```diff
@@ -70,6 +81,47 @@ router
...
+  })
+  .delete('/no_shows/:uuid', isUserAuthenticated, async (req, res, next) => {
+    try {
+      const { access_token, refresh_token } = req.user;
+      const { uuid } = req.params;
+      const calendlyService = new CalendlyService(access_token, refresh_token);
+
+      await calendlyService.undoNoShow(uuid);
+
+      res.status(204).send('No-show successfully undone');
```
Comment (tristanoneil) [inline routes/api.js:107]: I think we can just send an empty body response here. So you should be able to replace this with:

```js
res.status(204).end()
```

File: routes/api.js [hunk 3]
```diff
@@ -70,6 +81,47 @@ router
...
+    } catch (error) {
+      next(error);
+    }
+  })
+  .get('/no_shows/:uuid', isUserAuthenticated, async (req, res, next) => {
```
Comment (tristanoneil) [inline routes/api.js:112]: I don't _think_ we're using this endpoint?

**Review by tristanoneil — CHANGES_REQUESTED** [review]
Generally looks good, just some minor suggestions.

File: components/event.jsx [hunk 4]
```diff
@@ -35,5 +35,6 @@ export default () => {
                   </ul>
               )) : 'No guests added by invitee'}</div>
         </div>
```
[the remark this answers is not in the document]
Comment (RSalaam) [reply components/event.jsx:37]: Fixed! My next push will be good for all files I touch. Sorry about that!

File: components/invitee.jsx [hunk 5]
```diff
@@ -2,60 +2,114 @@ import React, { useState, useEffect } from 'react';
...
+      body: body,
+    });
+
+    fetchData();
+  };
+
+  const undoNoShowClick = async (event) => {
+    event.preventDefault();
+
+    const filteredInvitees = invitees.filter(
```
[the remark this answers is not in the document]
Comment (RSalaam) [reply components/invitee.jsx:43]: Ah, great idea. My next push will have this.

File: components/invitee.jsx (hunk 1 above)
Comment (RSalaam) [reply components/invitee.jsx:76]: Indeed. I meant to change this! Thank you for pointing this out.

File: routes/api.js (hunk 3 above)
Comment (RSalaam) [reply routes/api.js:112]: Correct. I had this to only allow no-show undoing if the invitee is, indeed, a no-show. I've worked this out via the UI where the undo button only shows if invitee is a no-show. Deleting! Also deleting the "findNoShow" method.

**Comment by RSalaam** [comment]
Forgot to mention that I updated pagination in the invitee component. Let me know if any issues/objections!

**Review by tristanoneil — APPROVED** [review]

**Pull request merged**